### PR TITLE
Scaffold multi-package workspace.

### PR DESCRIPTION
**Pull request by anthonyshew**
## Description

Create the structure for a multi-package workspace for the Node.js website.

Core maintainers have expressed a desire for re-using code from the website in other places. These changes allows for those publishable packages to be created, since they can now be broken out to their own separate packages.

Unfortunately, this involves a large-scale cutover from the repository's previous structure, invalidating PR's that come from current `main`. I've received maintainer approval for my disruptiveness. 🙏

Fortunately, those outstanding PRs can be fairly easily re-aligned with this PR, since I've made near-zero changes as far as the source code is concerned. This PR updates tooling, rather than functional code. (Prettier updated ~10 files that I saw at one point, but I don't believe this came from the changes I've made. I think the source code was somehow out-of-sync with the Prettier configuration.) If you'd like your PR to meet back up with this one, you should be able to move them from their current location to `./apps/site/<your-path>`.

## Validation

- One thing I can say I certainly don't know about in this repo is `crowdin.yml`. I'm not familiar with that tool so I don't know if it makes sense to move into the `apps/site` directory or if it needs to stay in the root.
- I've tried to find breakages and configuration differences from `main` to here and have satisfied my initial attempts to break myself.
- We can name the directory for the site something other than `site`, as desired. Just let me know! 
- I'll need some help making sure that CI has been re-aligned properly, both in GHA and in Vercel (cc @ovflowd in particular for Vercel, I believe).

## Anthony's Checklist

- [x] Does the site build locally?
- [x] Does husky pass?
- [x] Do lints pass locally?
- [x] Does lint-staged work?
- [x] Do test pass locally?
- [x] Do commands in the root of the repository run?
- [x] Do the expected `turbo` tasks run when invoked?
- [x] Update repo's meta-documentation for contributors
- [x] Does internationalization work as expected?
- [x] Does Vercel build?
- [ ] Re-align GHA checks

### CODE REVIEW

File: .github/workflows/lint-and-tests.yml [hunk 1]
```diff
@@ -45,9 +45,8 @@ jobs:
         # This step is responsible for providing a reusable string that can be used within other steps and jobs
         # that use the `turbo` cli command as a way of easily providing shared arguments to the `turbo` command
         id: turborepo_arguments
-        # See https://turbo.build/repo/docs/reference/command-line-reference/run#--cache-dir
         # See https://turbo.build/repo/docs/reference/command-line-reference/run#--force
-        run: echo "turbo_args=--force=true --cache-dir=.turbo/cache" >> "$GITHUB_OUTPUT"
```
Comment (anthonyshew) [inline .github/workflows/lint-and-tests.yml:50]: This is no longer needed as of Turborepo 2, since it's the default.

**Comment by bmuenzenmeyer** [comment]
I looked into crowdin renames - this was the best I could find so far.

https://community.crowdin.com/t/change-source-file-name-without-loosing-progress/7594/3

I will ask in our Slack channel with Crowdin to confirm or supply something more detailed

**Comment by Andrulko** [comment]
> One thing I can say I certainly don't know about in this repo is crowdin.yml. I'm not familiar with that tool so I don't know if it makes sense to move into the apps/site directory or if it needs to stay in the root.

Please keep this file in the root of the branch, it is used by Crowdin <-> GitHub connector:
https://support.crowdin.com/github-integration/

It would be required to update paths to EN files to be able to sync sources and translations:
https://github.com/nodejs/nodejs.org/blob/main/crowdin.yml

Please pause Crowdin <-> GitHub sync before merging the PR, then after it's merged Crowdin team will help to adapt the existing Crowdin project and configuration file to the new structure.

**Comment by anthonyshew** [comment]
Thanks, @Andrulko! I've remapped those paths now, if you'd like to check them to confirm I've done so correctly.

**Comment by Andrulko** [comment]
@anthonyshew Just to make sure that everything would work fine, let's insert `/` at the beginning of path. Please use the configuration below:

```
commit_message: 'chore: synced translations from crowdin [skip ci]'
append_commit_message: false
pull_request_title: '[automated]: crowdin sync'
pull_request_labels:
  - 'github_actions:pull-request'
files:
  - source: /apps/site/pages/en/**/*.md
    translation: /apps/site/pages/%two_letters_code%/**/%original_file_name%
    content_segmentation: 0
    ignore:
      - /apps/site/pages/en/blog/**/*.md
      - /apps/site/pages/en/learn/**/*.md
      - /apps/site/pages/en/download/index.md
      - /apps/site/pages/en/download/current.md
    languages_mapping:
      two_letters_code:
        es-ES: es
        pt-BR: pt-br
        zh-CN: zh-cn
        zh-TW: zh-tw
  - source: /apps/site/pages/en/**/*.mdx
    translation: /apps/site/pages/%two_letters_code%/**/%original_file_name%
    content_segmentation: 0
    ignore:
      - /apps/site/pages/en/blog/**/*.mdx
      - /apps/site/pages/en/learn/**/*.mdx
    languages_mapping:
      two_letters_code:
        es-ES: es
        pt-BR: pt-br
        zh-CN: zh-cn
        zh-TW: zh-tw
  - source: /apps/site/i18n/locales/en.json
    translation: /apps/site/i18n/locales/%two_letters_code%.json
    languages_mapping:
      two_letters_code:
        es-ES: es
        pt-BR: pt-br
        zh-CN: zh-cn
        zh-TW: zh-tw
```

Everything else looks good! Before you merge this branch, please pause GitHub sync in Crowdin (otherwise, it will upload new /apps/site folders along with files). I will help to move the existing files in Crowdin project to the new path and then we can safely resume the sync. No new files will be uploaded in this case 👌

**Comment by anthonyshew** [comment]
Updated with that. Thank you!

**Comment by bmuenzenmeyer** [comment]
## Windows Testing

✅ `npm run dev`
✅ `npm run build`
✅ `npm run lint`
❌ `npm run format`

```
PS C:\src\nodejsorg> npm run format

> format
> npm run lint:fix && npm run prettier:fix


> lint:fix
> turbo run lint --force -- --fix

╭──────────────────────────────────────────────────────────────────────╮
│                                                                      │
│                   Update available v2.0.3 ≫ v2.0.5                   │
│    Changelog: https://github.com/vercel/turbo/releases/tag/v2.0.5    │
│           Run "npx @turbo/codemod@latest update" to update           │
│                                                                      │
│        Follow @turborepo for updates: https://x.com/turborepo        │
╰──────────────────────────────────────────────────────────────────────╯
• Packages in scope: @nodejs/website
• Running lint in 1 packages
• Remote caching disabled
┌ @nodejs/website#lint > cache bypass, force executing e6bbf3daeb3b7be9
│
│ > lint
│ > turbo run lint:md lint:js lint:css --fix
│
│ ╭──────────────────────────────────────────────────────────────────────╮
│ │                                                                      │
│ │                   Update available v2.0.3 ≫ v2.0.5                   │
│ │    Changelog: https://github.com/vercel/turbo/releases/tag/v2.0.5    │
│ │           Run "npx @turbo/codemod@latest update" to update           │
│ │                                                                      │
│ │        Follow @turborepo for updates: https://x.com/turborepo        │
│ ╰──────────────────────────────────────────────────────────────────────╯
│  ERROR  unexpected argument '--fix' found
│
│   tip: to pass '--fix' as a value, use '-- --fix'
│
│ Usage: turbo.exe run <--cache-dir <CACHE_DIR>|--concurrency <CONCURRENCY>|--continue|--singl
│ e-package|--force [<FORCE>]
│ |--framework-inference [<BOOL>]|--global-deps <GLOBAL_DEPS>|--env-mode [<ENV_MODE>]|--filter
│  <FILTER>|--output-logs <OU
│ TPUT_LOGS>|--log-order <LOG_ORDER>|--only|--pkg-inference-root <PKG_INFERENCE_ROOT>|--remote
│ -only [<BOOL>]|--log-prefix
│  <LOG_PREFIX>|TASKS|PASS_THROUGH_ARGS>
│
│ For more information, try '--help'.
│
│ npm ERR! Lifecycle script `lint` failed with error:
│ npm ERR! Error: command failed
│ npm ERR!   in workspace: @nodejs/website
│ npm ERR!   at location: C:\src\nodejsorg\apps\site
│ command finished with error: command (C:\src\nodejsorg\apps\site) C:\Program Files\nodejs\np
│ m.cmd run lint -- --fix exited (1)
└────>
┌ @nodejs/website#lint:css > cache bypass, force executing c5eb2653016c9fd1
│
│ > lint:css
│ > stylelint "**/*.css" --allow-empty-input --cache --cache-strategy=content --cache-location
│ =.stylelintcache
└────>
┌ @nodejs/website#lint:js > cache bypass, force executing d4e4c6408b1b624b
│
│ > lint:js
│ > eslint "**/*.{js,mjs,ts,tsx}" --cache --cache-strategy=content --cache-location=.eslintjsc
│ ache
└────>
┌ @nodejs/website#lint:md > cache bypass, force executing 0a254661a087c88d
│
│ > lint:md
│ > eslint "**/*.md?(x)" --cache --cache-strategy=content --cache-location=.eslintmdcache
└────>
 ERROR  run failed: command  exited (1)
```

File: package.json [hunk 2]
```diff
@@ -16,119 +16,30 @@
...
-    "lint:md": "eslint \"**/*.md?(x)\" --cache --cache-strategy=content --cache-location=.eslintmdcache",
-    "lint:css": "stylelint \"**/*.css\" --allow-empty-input --cache --cache-strategy=content --cache-location=.stylelintcache",
-    "lint": "turbo run lint:md lint:js lint:css",
-    "lint:fix": "turbo run lint:md lint:js lint:css --no-cache -- --fix",
+    "dev": "turbo run dev",
+    "build": "turbo run build",
+    "start": "turbo run start",
+    "check-types": "turbo check-types",
+    "lint": "turbo run lint",
+    "lint:fix": "turbo run lint --force -- --fix",
```
Comment (bmuenzenmeyer) [inline package.json]: passing the argument another level down worked for me - testing on other platforms should occur.
```suggestion
    "lint:fix": "turbo run lint --force -- -- --fix",
```

perhaps: can we add formatting to turbo's tasks?

Comment (anthonyshew) [reply package.json]: Oops! Good find. Thank you, fixed that.

I had the same error on macOS and fixed it up now. Not sure if you want further platform validation.

**Comment by ovflowd** [comment]
cc @anthonyshew some of the CI steps are still failing

**Comment by anthonyshew** [comment]
Yep, working on those right now!

**Comment by ovflowd** [comment]
@anthonyshew lmk when you believe this PR is ready.

Also don't forget to fix the sync orama script and ensure the blog post generation one is also working!

**Comment by bmuenzenmeyer** [comment]
I've paused Crowdin Sync and made a discussion post over there to describe the changes.

**Comment by anthonyshew** [comment]
I _think_ I've gotten things as close as I can at this point without giving it another real run:
- Squashed everything down to one commit for rebasing purposes
- Hand-tested the blog post generation and Orama syncing scripts
- Ran [tests](https://github.com/anthonyshew/nodejs.org/actions/runs/9784684619/job/27016117748?pr=1) and [lints](https://github.com/anthonyshew/nodejs.org/actions/runs/9784684619/job/27016117980?pr=1) on my fork to see them green
- Updated the working directory for the Storybook Action

With that said, I think I'm ready to get labelled for another run again. 🙏

**Comment by ovflowd** [comment]
Thanks, @anthonyshew! Your work and effort is extremely appreciated!

I believe CI will fail in this branch but we can proceed with a merge if you are confident with things.

cc @nodejs/nodejs-website and @bmuenzenmeyer for a final PR review run 👀

**Comment by anthonyshew** [comment]
Looking through the CI errors on this run, these are all what I would expect to see from the changes I've made running against the Actions from `main`. I _think_ I've accounted for all the failures I'm seeing but am prepared to be pseudo-on-call and quickly fix-forward as needed.

The Root Directory on Vercel will need to be updated to `apps/site`, which would earn these checks another green check mark if re-deployed with that setting. Even if not, you'd want to update that setting prior to merging, so that you get a good production deployment.

**Comment by ovflowd** [comment]
Gotcha! Ill give another review in a few minutes and Im fine with bearing the responsibilities of the merge.

**Review by ovflowd — APPROVED** [review]
It looks good to me. @bmuenzenmeyer note that Crowdin might break? But that's an issue to verify as a follow-up. I'm merging this now and going to observe what happens on the main branch

**Pull request merged**